Error in array index; 0 not in range [0..0)
Function Even is not in scope
diff --git a/src/compiler/GF/Compile/GrammarToPGF.hs b/src/compiler/GF/Compile/GrammarToPGF.hs
index cc560ca..dbe5df4 100644
--- a/src/compiler/GF/Compile/GrammarToPGF.hs
+++ b/src/compiler/GF/Compile/GrammarToPGF.hs
@@ -277,7 +277,8 @@ genCncFuns gr am cm seqs0 cdefs fid_cnt cnccats =
                                           (seqs,(offs+funid0,C.CncFun (i2i id) lins0):funs)
       where
         mkLin mseqs seqs seqid =
-          let seq = mseqs ! seqid
+          let seq =  mseqs ! seqid
+              {-# INLINE seq #-}
           in case Map.lookup seq seqs of
                Just seqid -> (seqs,seqid)
                Nothing    -> let !seqid = Map.size seqs
